### PR TITLE
Homework2 : Nicolay Krischenovich

### PR DESCRIPTION
**Pull request by Nicolay-kr**

### CODE REVIEW

**Comment by Nicolay-kr** [comment]
Homework2

File: homework2_extratask1.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# циклические операторы и условные операторы. n - вводится
+
+f1 = 0
+f2 = f3 = 1
+n = int(input("Введите необходимое количество чисел Фибоначчи: "))
+print(f1, end=', ')
+print(f2, end=', ')
+print(f3, end=', ')
+for i in range(n):
+    f2, f3 = f3, f2 + f3
```
Comment (belovmd) [inline homework2_extratask1.py:11]: great

File: homework2_extratask2.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+s = int(input('Введите число: '))
+n = s
+m = 0
+i = 0
+while n // 10 ** i:
+    i += 1
+print(i)
+
+for j in range(1, i + 1):
```
Comment (belovmd) [inline homework2_extratask2.py]: it looks too complex - it can be easier solved with while loop

File: homework2_task2-1.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Simple, remove the spaces from the string, then return the resultant string.
+def no_space(x):
+    s = ""
+    for i in x:
```
Comment (belovmd) [inline homework2_task2-1.py]: replace?

File: homework2_task2-2.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+# For example if you are given a chocolate bar of size 2 x 1
+# you can split it to
+# single squares in just one break, but for size 3 x 1 you must do two breaks.
+# If input data is invalid you should return 0 (as in no breaks are needed
+# if we do not have any chocolate to split). Input will always be a
+# non-negative integer.
+
+
+def breakchocolate(n, m):
+    s = n * m
```
Comment (belovmd) [inline homework2_task2-2.py]: s n m i - bad naming

File: homework2_task2-2.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    q = 0
+    if s > 1:
+        for i in range(1, s):
```
Comment (belovmd) [inline homework2_task2-2.py]: for 3x3 you solution will give 8 breaks but in fact 6 would be enough

File: homework2_task3.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# Найти самое длинное слово в введенном предложении. Учтите что в предложении
+# есть знаки препинания.
+# Подсказки:
+# my_string.split([chars]) возвращает список строк.
+# len(list) - количество элементов в списке
+
+s = " Найти самое, длинное, слово в введенном: предложении."
+n = s.split()
+for i in range(len(n)):
+    n[i] = n[i].strip(".,!?:")
```
Comment (belovmd) [inline homework2_task3.py]: what about (){} etc? ' ''

File: homework2_task4.py [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# Вводится строка. Требуется удалить из нее повторяющиеся
+# символы и все пробелы.
+# Например, если было введено "abc cde def", то должно быть выведено "abcdef".
+s = "abc cde def prrr ggggg l a s d ffff c  112 33"
```
Comment (belovmd) [inline homework2_task4.py]: please add input

File: homework2_task5.py [hunk 8]
```diff
@@ -0,0 +1,16 @@
+# Посчитать количество строчных (маленьких) и прописных (больших) букв в
+# введенной строке. Учитывать только английские буквы.
+s = str(input('Введите строку: '))
+b = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
+c = 'abcdefghijklmnopqrstuvwxyz'
+n = ''
+m = ''
+for i in s:
```
Comment (belovmd) [inline homework2_task5.py:6]: islower() isupper()

**Review by belovmd — COMMENTED** [review]
1. Overall external tasks are pretty easy. you can try to solve some advanced tasks
2. Please address these issues

**Comment by Nicolay-kr** [comment]
Fix Homework 2

File: homework2_task2-1.py [hunk 9]
```diff
@@ -0,0 +1,11 @@
+# Simple, remove the spaces from the string, then return the resultant string.
+def no_space(x):
+    s = x.replace(' ', '')
+    # s = ""
+    # for i in x:
```
Comment (belovmd) [inline homework2_task2-1.py:5]: redundant comments

**Review by belovmd — COMMENTED** [review]
8

**Pull request closed without merging**